src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,102 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_SIZE = 10;
+    private T[] object;
+    private int size = 0;
+
+    public ArrayList() {
+        object = (T[]) new Object[DEFAULT_SIZE];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == object.length) {
+            increaseListSize();
+        }
+        object[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index exceeds size of the list");
+        }
+        if (size == object.length) {
+            increaseListSize();
+        }
+        System.arraycopy(object, index, object, index + 1, size - index);
+        object[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index < size && index >= 0) {
+            return object[index];
+        } else {
+            throw new ArrayListIndexOutOfBoundsException("Index passed is invalid");
+        }
     }
 
     @Override
     public void set(T value, int index) {
-
+        if (index < size && index >= 0) {
+            object[index] = value;
+        } else {
+            throw new ArrayListIndexOutOfBoundsException("Index passed is invalid");
+        }
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        if (index < size && index >= 0) {
+            T currentObject = object[index];
+            System.arraycopy(object, index + 1, object, index, size - index - 1);
+            object[--size] = null;
+            return currentObject;
+        } else {
+            throw new ArrayListIndexOutOfBoundsException("Index passed is invalid");
+        }
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        for (int i = 0; i < size; i++) {
+            if (Objects.equals(element, object[i])) {
```

Comment:
You shouldn't use class Objects in this task. [checklist](https://mate-academy.github.io/jv-program-common-mistakes/java-core/collections/array-list.html)